docs/mayastor-concept.md

```diff
@@ -0,0 +1,170 @@
+
+The instance of the `mayastor` binary running inside the container performs four major classes of functions:
+
+* Present a gRPC interface to the MOAC control plane component, to allow the latter to orchestrate creation, configuration and deletion of Mayastor managed objects hosted by that instance
+* Create and manage storage pools hosted on that node
+* Create, export and manage nexus objects \(and by extension, volumes\) hosted on that node
+* Create and share "replicas" from storage pools hosted on that node
+  * Local replica -&gt; loopback -&gt; Local Nexus
+  * Local replica - &gt; NVMe-F TCP -&gt;  Remote Nexus \(hosted by a Mayastor container on another node\)
+  * Remote replicas are employed by a Nexus as synchronous data copies, where replication is in use
+
+When a Mayastor pod starts running, an init container attempts to verify connectivity to the NATS message bus in the Mayastor namespace.  If a connection can be established the Mayastor container is started, and the Mayastor instance performs registration with MOAC over the message bus.  In this way, MOAC maintains a registry of nodes \(specifically, running Mayastor instances\) and their current state.  For each registered Mayastor container/instance, MOAC creates a MayastorNode custom resource within the Kubernetes API of the cluster.
+
+The scheduling of Mayastor pods is determined declaratively by using a DaemonSet specification.  By default, a `nodeSelector` field is used within the pod spec to select all worker nodes to which the user has attached the label `openebs.io/engine=mayastor` as recipients of a Mayastor pod.   It is in this way that the MayastorNode count and location is set appropriate to the hardware configuration of the worker nodes \(i.e. which nodes host the block storage devices to be used\), and capacity and performance demands of the cluster.
+
+#### Mayastor-CSI
+
+The mayastor-csi pods within a cluster implement the node plugin component of Mayastor's CSI driver.  As such, their function is to orchestrate the mounting of Maystor provisioned volumes on worker nodes on which application pods consuming those volumes are scheduled.  By default a mayastor-csi pod is scheduled on every node in the target cluster, as determined by a DaemonSet resource of the same name.  These pods each encapsulate two containers, `mayastor-csi` and `csi-driver-registrar`
+
+ It is not necessary for the node plugin to run on every worker node within a cluster and this behaviour can be modified if so desired through the application of appropriate node labeling and the addition of a corresponding  `nodeSelector` entry within the pod spec of the mayastor-csi DaemonSet.  It should be noted that if a node does not host a plugin pod, then it will not be possible to schedule pod on it which is configured to mount Mayastor volumes.
```

Comment:
```suggestion
 It is not necessary for the node plugin to run on every worker node within a cluster. This behavior can be modified if so desired by applying appropriate node labeling and the addition of a corresponding  `nodeSelector` entry within the pod spec of the mayastor-csi DaemonSet.  If a node does not host a plugin pod, it will not be possible to schedule a pod on it which is configured to mount Mayastor volumes.
```